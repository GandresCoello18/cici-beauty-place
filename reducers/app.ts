const SET_TITLE = 'app/set-title'

export const setTitle = (v: string) => ({
  type: SET_TITLE as typeof SET_TITLE,
  payload: v,
})

export type AppAction = ReturnType<typeof setTitle>

export const initialState = {
  title: '',
}

export type AppState = typeof initialState

export const reducer = (state = initialState, action: AppAction) => {
  if (action.type === SET_TITLE) {
    return { ...state, title: action.payload }
  }
  return state
}
