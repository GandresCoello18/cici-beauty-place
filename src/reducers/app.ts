const SET_TITLE = 'app/set-title'

export const setTitle = (v: string) => ({
  type: SET_TITLE as typeof SET_TITLE,
  payload: v,
})

export type AppAction = ReturnType<typeof setTitle>

export type AppState = typeof initialState

export const initialState = {
  title: '',
}

export const reducer = (state = initialState, action: AppAction) => {
  if (action.type === SET_TITLE) {
    return { ...state, title: action.payload }
  }
  return state
}
