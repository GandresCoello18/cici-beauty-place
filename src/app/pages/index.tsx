import { Store } from 'redux'
import React from 'react'
import { App } from '../containers/App'
import { AllAction, RootState } from '../reducers'
import { setTitle } from '../reducers/app'

const Index = () => <App />

Index.getInitialProps = async ({
  store,
}: {
  store: Store<RootState, AllAction>
}) => {
  // redux
  store.dispatch(setTitle('Server Side Rendering'))
}

export default Index
