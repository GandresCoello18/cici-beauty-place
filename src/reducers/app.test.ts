import { reducer, setTitle } from './app'

test('should return the initial state', () => {
  expect(
    reducer(undefined, { type: undefined as any, payload: undefined as any })
  ).toMatchSnapshot()
})

test('should handle SET_TITLE', () => {
  expect(
    reducer(undefined, setTitle('Client Side Rendering'))
  ).toMatchSnapshot()
})
