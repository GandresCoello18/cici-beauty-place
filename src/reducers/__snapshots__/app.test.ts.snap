// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle SET_TITLE 1`] = `
Object {
  "title": "Client Side Rendering",
}
`;

exports[`should return the initial state 1`] = `
Object {
  "title": "",
}
`;
